protobufs/livekit_sip.proto

```diff
@@ -132,16 +147,90 @@ message SIPTrunkInfo {
   // User-defined metadata for the Trunk.
   string metadata = 12;
 
-  // NEXT ID: 14
+  // NEXT ID: 15
+}
+
+message CreateSIPInboundTrunkRequest {
+  SIPInboundTrunkInfo trunk = 1; // Trunk ID is ignored
+}
+
+message SIPInboundTrunkInfo {
+  string sip_trunk_id = 1;
+
+  // Human-readable name for the Trunk.
+  string name = 2;
+  // User-defined metadata for the Trunk.
+  string metadata = 3;
+
+  // Numbers associated with LiveKit SIP. The Trunk will only accept calls made to these numbers.
+  // Creating multiple Trunks with different phone numbers allows having different rules for a single provider.
+  repeated string numbers = 4;
+
+  // CIDR or IPs that traffic is accepted from.
+  // An empty list means all inbound traffic is accepted.
+  repeated string allowed_addresses = 5;
+
+  // Numbers that are allowed to make calls to this Trunk.
+  // An empty list means calls from any phone number is accepted.
+  repeated string allowed_numbers = 6;
+
+  // Username and password used to authenticate inbound SIP invites.
+  // May be empty to have no authentication.
+  string auth_username = 7;
+  string auth_password = 8;
+}
+
+message CreateSIPOutboundTrunkRequest {
+  SIPOutboundTrunkInfo trunk = 1; // Trunk ID is ignored
+}
+
+message SIPOutboundTrunkInfo {
+  string sip_trunk_id = 1;
+
+  // Human-readable name for the Trunk.
+  string name = 2;
+  // User-defined metadata for the Trunk.
+  string metadata = 3;
+
+  // Hostname or IP that SIP INVITE is sent too.
+  // Note that this is not a SIP URI and should not contain the 'sip:' protocol prefix.
+  string address = 4;
+
+  // SIP Transport used for outbound call.
+  SIPTransport transport = 5;
+
+  // A number used to make the calls.
+  string number = 6;
```

Comment:
On the trunking provider side, it's still possible to associate multiple numbers to the SIP trunk. but I guess the point here is when making a call, a particular number will always be used?

Reply:
Yes, it's possible to use multiple numbers for outbound on the provider's side.

Exactly, this is to specify which specific number to use. Having said that, I was also considering an array here. The implementation could pick a random number from an array before making a call.

Lets allow multiple outbound numbers then.

Reply:
so maybe for the protocol, it'd be better to leave it as a repeated field? but in usage, we would let people know that only the first is used?

in the future, we can then add ways of choosing a trunk at run-time

Reply:
Already updated to repeated field :+1: It will actually use the random one from the list if the Trunk is created with new API. And if not - it will be the same as before (one number only).